libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -91,11 +95,18 @@ fun NavigationRoute.refreshRoute(
                 .build()
         }
     }
-    return updateDirectionsRouteOnly {
-        toBuilder()
-            .legs(updateLegs)
-            .updateRouteDurationBasedOnLegsDuration(updateLegs)
-            .build()
+    return update(
+        {
+            toBuilder()
+                .legs(updateLegs)
+                .updateRouteDurationBasedOnLegsDuration(updateLegs)
+                .build()
+        }
+    ) {
+        updateWaypoints(
+            directionsResponse.waypoints(),
+            unrecognizedProperties
+        )
     }
```

Comment:
```suggestion
        directionsRouteBlock = {
            toBuilder()
                .legs(updateLegs)
                .updateRouteDurationBasedOnLegsDuration(updateLegs)
                .build()
        },
        directionsResponseBlock = {
            updateWaypoints(
                directionsResponse.waypoints(),
                unrecognizedProperties
            )
        }
    )
```
nit, to improve readability